deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -454,6 +436,38 @@ def _set_dtype(self, ds_config, dtype):
             self.dtype = dtype or torch.half
 
 
+def shutdown_init_context():
+    global zero_init_enabled
+
+    if not zero_init_enabled:
+        return
+
+    def _disable_class(cls):
+        cls.__init__ = cls._old_init
```

Comment:
Looking at the errors in your CI tests, it looks like it is trying to disable the class when it wasn't enabled before:
`AttributeError: type object 'HingeEmbeddingLoss' has no attribute '_old_init'`

A reason for this could be (just a guess) that the `zero_init_enabled` global variable somehow remains True in one test and then leaks into another.

Reply:
I think this issue is resolved now, waiting on all the tests to run though.